### ISSUE TITLE
forbidden local links pointing to files outside of the build directory

### ISSUE THREAD

**Issue by ThomasBreuer** [JuliaDocs/Documenter.jl#2331]
From version 1.0 of Documenter.jl on, an `Error: invalid local link/image: path pointing to a file outside of build directory` is thrown if a local cross-reference points somewhere else.
Earlier versions of Documenter.jl did not forbid such cross-references.

In the documentation of GAP.jl, we want to insert cross-references to another manual, pointing to somewhere in `assets/html`, where the relevant pages are available.
What is wrong with such cross-references? How can we keep them when we use the new Documenter.jl?
(See oscar-system/GAP.jl/pull/953 for the current situation.)

**Comment by mortenpi** [comment]
I'm not quite sure where your `assets/` are coming from, but I assume that you have them in `docs/build/assets/` somehow. The correct relative link should be e.g. `assets/html/JuliaInterface/chap0_mj.html` for any `.md` files directly in `src/` (as opposed to subdirectories). In other words, [this link should work](https://github.com/oscar-system/GAP.jl/pull/953/files#diff-36c852523b6a7513a10b8a6bae21f1037eaa27484efc9042f75bbda46d9f529bL33) and you should also remove the `../` from the other links.

It's _possible_ that this is still somehow buggy, so let me know if it's still not working, and I'll see if I can repro.

As a note though: no, Documenter is not designed to pick up stuff from outside of the `src` / `build` directories.

**Comment by ThomasBreuer** [comment]
@mortenpi Thanks for the hint, now the documentation in question can be processed, provided that the files from the documentation to the "other" package are already available and can be copied in advance into the `src` directory.

Note that this would be impossible if the documentation of the "other" package would contain local back-references and if this documentation would be built with a tool that requires the target files to be available in advance, like Documenter does.
In other words, I think it is too restrictive to expect that all target files exist already during the build process.

**Comment by fingolfin** [comment]
@ThomasBreuer well to be fair, in the worst case we could always just `touch` those `.html` files until the final ones are available...

@mortenpi Thanks for the advise! Removing the `../` worked -- we assumed these were necessary (because of the nesting of the resulting final web pages), and the fact it worked (or seemed to at least) confirmed that -- but I guess only "by accident"? Anyway, happy this is now resolved :-)

**Comment by mortenpi** [comment]
> In other words, I think it is too restrictive to expect that all target files exist already during the build process.

Well, this enables a very useful check 90% of the time, and your use case seems a bit niche / unusual. The workaround suggested by @fingolfin seems pretty reasonable in that situation. Of course, we could also consider adding some configuration options to ignore certain paths etc., but I am not the use case is common enough to warrant that.

It seems like the main issue was solved, so let's close this!
